### PR TITLE
Document org name input & use new validator

### PR DESCRIPTION
**Pull request by aredridel**
Note that this input sets the scope name rather than the display name (for now) 
- [x] add each character as the user types
- [x] if the user clears the field, set it to "your-org-name"
- [x] Provide errors and add validations for username restrictions: can't start with a period, no uppercase, char limit. See the following for details: https://github.com/npm/npm-user-validate/blob/master/npm-user-validate.js#L17-L31
- [x] ~~maybe use https://www.npmjs.com/package/slug ? :-)~~

[#119417073]

### CODE REVIEW

**Comment by aredridel** [comment]
screen shot 2016-06-09 at 18 15 25

**Comment by aredridel** [comment]
Shown with validation state.

**Comment by aredridel** [comment]
Per @jerrygabra :

> Let it be the same as story #119563151 but with different label and notification text.
> Errors appear underneath the notification text, use the warning style from PUI.

**Comment by aredridel** [comment]
Initial state screen shot 2016-06-09 at 18 20 43

File: src/pivotal-ui/components/forms/link-updater.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ var LinkUpdater = function(el){
 LinkUpdater.prototype.updateValue = function(inputValue){
   var pathVal = inputValue;
   if(inputValue === ""){
-    pathVal = "username";
+    pathVal = this.input.attr("placeholder") || "username";
```
Comment (jefflembeck) [inline src/pivotal-ui/components/forms/link-updater.js]: Let's set this on initialization. Way easier for testing.

Comment (aredridel) [inline src/pivotal-ui/components/forms/link-updater.js]: Interesting. Means that the association isn't dynamic, though -- it means updating the placeholder won't be reflected.

Comment (jefflembeck) [inline src/pivotal-ui/components/forms/link-updater.js]: Maybe I'm being ridiculous here, but I think that might be a YAGNI case.

Comment (aredridel) [inline src/pivotal-ui/components/forms/link-updater.js]: Fair enough!

**Comment by aredridel** [comment]
https://github.com/npm/npm-website/blob/94c758008cafb0137b61ad0173b8ca851461d332/test-frontend/org-name-input.js is a unit test for this JS

**Comment by aredridel** [comment]
Oh good call. Error state test at the unit level:

https://github.com/npm/npm-website/blob/aredridel/org-name-input/test-frontend/org-name-input.js

**Comment by gabrrra** [comment]
Thanks @aredridel — this looks good!

**Pull request merged**